### PR TITLE
Provide descriptive statistics for supreme corpus data

### PR DESCRIPTION
**Pull request by chanteriam**
## Describe your changes
Added a descriptive statistics class for the supreme corpus data.

## Non-obvious technical information
N/A

## Checklist before requesting a review

- [x] `make format` and `make lint` have been run and the output has been addressed.
- [x] The code runs successfully.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -16,6 +16,9 @@ responses = "^0.23.1"
 convokit = "^2.5.3"
 pandas = "^2.0.0"
 isort = "^5.12.0"
+nltk = "^3.8.1"
+sklearn = "^0.0.post4"
+openpyxl = "^3.1.2"
```
Comment (michplunkett) [inline pyproject.toml:21]: Noiiiice.

File: supreme_court_predictions/statistics/descriptives.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        if not downloaded_clean_corpus:
+            print("Cleaned corpus not present, downloading...")
+            cleaner = DataCleaner(True)
+            cleaner.parse_all_data()
+        else:
+            print("Cleaned corpus already downloaded")
+
+    def get_count_desc(self, df, cols):
+        """
+        [Insert]
```
Comment (michplunkett) [inline supreme_court_predictions/statistics/descriptives.py:55]: You got some of these bois to fill out.

File: supreme_court_predictions/statistics/descriptives.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+        print("Grabbing voters descriptive statistics...")
+        self.voters_stats = self.get_voters_desc()
+
+        if self.save_data:
+            desc_out = self.output_path + "/descriptive_statistics.xlsx"
+            self.advocates_stats.to_csv(
+                self.output_path + "/advocates_stats.csv",
+                index=True,
+                encoding=ENCODING_UTF_8_SIG,
```
Comment (michplunkett) [inline supreme_court_predictions/statistics/descriptives.py:217]: Do you need the `utf-8 sig` value or just `utf-8`?

File: .gitignore [hunk 4]
```diff
@@ -4,10 +4,14 @@
...
-/supreme_court_predictions/data/convokit/*
-!/supreme_court_predictions/data/convokit/README.md
 /supreme_court_predictions/data/clean_convokit/*
 !/supreme_court_predictions/data/clean_convokit/README.md
+/supreme_court_predictions/data/convokit/*
+!/supreme_court_predictions/data/convokit/README.md
+/supreme_court_predictions/data/models/*
+!/supreme_court_predictions/data/models/README.md
+/supreme_court_predictions/data/statistics/*
+!/supreme_court_predictions/data/statistics/README.md
```
Comment (michplunkett) [inline .gitignore:14]: Updated the `data` ignores.

File: supreme_court_predictions/statistics/descriptives.py [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+
+import numpy as np
+import pandas as pd
+
+from supreme_court_predictions.util.contants import ENCODING_UTF_8_SIG
+
+from .datacleaner import DataCleaner
+
+
+class Descriptives:
```
Comment (michplunkett) [inline supreme_court_predictions/statistics/descriptives.py:15]: Could you also put the calling of this class in the `/supreme-court-ml-predictions/supreme_court_predictions/statistics/services.py` file?

**Review by FedericoDM — APPROVED** [review]
Greaaat <3

File: supreme_court_predictions/statistics/descriptives.py (hunk 2 above)
Comment (chanteriam) [reply supreme_court_predictions/statistics/descriptives.py:55]: I know michael i am lazy, I'll do it tomorrow :p

File: supreme_court_predictions/statistics/descriptives.py (hunk 3 above)
Comment (chanteriam) [reply supreme_court_predictions/statistics/descriptives.py:217]: i have no idea, i was copying fede's code @FedericoDM

File: supreme_court_predictions/statistics/descriptives.py (hunk 5 above)
Comment (chanteriam) [reply supreme_court_predictions/statistics/descriptives.py:15]: sure can!

File: supreme_court_predictions/statistics/descriptives.py (hunk 3 above)
Comment (michplunkett) [reply supreme_court_predictions/statistics/descriptives.py:217]: > The Unicode Standard permits the BOM in [UTF-8](https://en.wikipedia.org/wiki/UTF-8),[[3]](https://en.wikipedia.org/wiki/Byte_order_mark#cite_note-3) but does not require or recommend its use.[[4]](https://en.wikipedia.org/wiki/Byte_order_mark#cite_note-4)
- https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8

It looks like we should roll with the other UTF-8 instead. Let me know if I'm missing something, @FedericoDM.

File: supreme_court_predictions/statistics/descriptives.py [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+            for idx, desc in enumerate(descriptives):
+                desc.to_csv(outpaths[idx], index=True, encoding=ENCODING_UTF_8)
+
+            # Outputting to a single excel
+            desc_out = self.output_path + "/descriptive_statistics.xlsx"
+            # Known ExcelWriter issue:
+            # https://github.com/pylint-dev/pylint/issues/3060
+            # pylint: disable=abstract-class-instantiated
+            with pd.ExcelWriter(desc_out) as writer:
+                # pylint: enable=abstract-class-instantiated
```
Comment (michplunkett) [inline supreme_court_predictions/statistics/descriptives.py:315]: I added these comments to escape a linting issue: https://github.com/pylint-dev/pylint/issues/3060

cc: @chanteriam

**Review by michplunkett — APPROVED** [review]
This is fantastic. Well done, @chanteriam!

**Pull request merged**